src/Components/Server/test/Circuits/CircuitRegistryTest.cs

```diff
@@ -678,4 +689,148 @@ private static (CircuitRegistry Registry, TestCircuitPersistenceProvider Provide
             persistenceManager);
         return (registry, provider);
     }
+
+    [Fact]
+    public async Task PauseAfterTermination_DoesNotThrow()
+    {
+        var circuitIdFactory = TestCircuitIdFactory.CreateTestFactory();
+        var options = new CircuitOptions();
+
+        var circuitHost = new TestCircuitHostForRaceConditions(
+            circuitIdFactory.CreateCircuitId(),
+            CreateServiceScope(),
+            options);
+
+        var persistenceProvider = new TestCircuitPersistenceProvider();
+        var registry = new TestCircuitRegistry(circuitIdFactory, options, persistenceProvider);
+        registry.Register(circuitHost);
+
+        // First terminate the circuit - it calls circuitHost.DisposeAsync()
+        await registry.TerminateAsync(circuitHost.CircuitId);
+
+        // Then try to pause - it will try to resolve services from the DI scope that is already disposed
+        await registry.PauseAndDisposeCircuitHost(circuitHost, saveStateToClient: true);
+    }
+
+    [Fact]
+    public async Task MultiplePause_DoesNotThrow()
+    {
+        var circuitIdFactory = TestCircuitIdFactory.CreateTestFactory();
+        var options = new CircuitOptions();
+
+        var circuitHost = new TestCircuitHostForRaceConditions(
+            circuitIdFactory.CreateCircuitId(),
+            CreateServiceScope(),
+            options);
+
+        var persistenceProvider = new TestCircuitPersistenceProvider();
+        var registry = new TestCircuitRegistry(circuitIdFactory, options, persistenceProvider);
+        registry.Register(circuitHost);
+
+        // First pause should be successful
+        await registry.PauseAndDisposeCircuitHost(circuitHost, saveStateToClient: true);
+
+        // Second pause - it will try to resolve services from the DI scope that is already disposed
+        await registry.PauseAndDisposeCircuitHost(circuitHost, saveStateToClient: true);
+    }
```

Comment:
We shouldn't have this test. Calling pause multiple times is not a thing we expect to happen, we shouldn't test for it, we should preclude it from happening.